go/fixchain/fixer.go

```diff
@@ -24,20 +24,19 @@ type Fixer struct {
 	notReconstructed uint
 	fixed            uint
 	notFixed         uint
-	skipped          uint
-	alreadyDone      uint
 
 	wg    sync.WaitGroup
 	cache *urlCache
-	done  *lockedMap
 }
 
 // QueueChain adds the given cert and chain to the queue to be fixed by the
-// fixer, with respect to the given roots.
+// fixer, with respect to the given roots.  Note: chain is expected to be in the
+// order of cert --> root.
 func (f *Fixer) QueueChain(cert *x509.Certificate, chain []*x509.Certificate, roots *x509.CertPool) {
+	dchain := newDedupedChain(chain)
 	f.toFix <- &toFix{
 		cert:  cert,
-		chain: newDedupedChain(chain),
+		chain: dchain,
```

Comment:
No idea :)  I'd imagine I considered putting something else in there then backtracked but never made it all the way back.  Fixed now.